pkg/serverless/daemon/daemon.go

```diff
@@ -435,3 +447,17 @@ func (d *Daemon) setTraceTags(tagMap map[string]string) bool {
 	}
 	return false
 }
+
+// IsLambdaLibraryDetected returns if the Lambda Library is in use
+func (d *Daemon) IsLambdaLibraryDetected() bool {
+	d.LambdaLibraryStateLock.RLock()
+	defer d.LambdaLibraryStateLock.RUnlock()
+	return d.LambdaLibraryDetected
+}
+
+// IsExecutionSpanComplete checks if the execution span was finished
+func (d *Daemon) IsExecutionSpanComplete() bool {
+	d.ExecutionSpanStateLock.Lock()
+	defer d.ExecutionSpanStateLock.Unlock()
+	return d.hitOnStart && d.hitOnEnd
```

Comment:
Where are these fields set back to false again? From what I am seeing, once they become true, they will never become false.

Reply:
Ah good catch, my mistake. Also makes me realize we probably only need one bool to keep track of whether the execution span was finished so both the `hitOnStart` + `hitOnEnd` feels unnecessary. Working on this